### ISSUE TITLE
Forcing deployments behavior is poorly explained

### ISSUE THREAD

**Issue by meichstedt** [mesosphere/marathon#3062]
tldr; the need for and results of stopping a deployment when trying to trigger a new deployment are poorly described and imho misleading to the user. Even the service docs do not completely explain what stopping a deployment means.

The error dialog resulting in trying to trigger a deployment for an application that is currently affected by a running deployment was changed in #2841.

When I now trigger a deployment for an app that is currently affected by a deployment, I get an error message stating

> Error Restarting Application
> In order to restart /asd, the current deployment will have to be forcefully stopped. Please be cautious, as this could result in unwanted states.

Originating from the [AppActionsHandlerMixin.jsx](https://github.com/mesosphere/marathon-ui/blob/master/src/js/mixins/AppActionsHandlerMixin.jsx#L323)

I have several problems with this message
1. _In order to restart ... will have to be ..._ seems misleading to me. Stopping the current deployment is not a prerequisite for a restart. The deployment has to be _somehow_ terminated to _enable_ a new deployment of any kind (sorry for nit picking here). The message could also say _In order to restart ... you have to wait for the current deployment to be finished_ (which would make more sense if the current deployment is not stuck in any way).
2. _Please be cautious_ is even more misleading. The user is not able to apply any caution when stopping a deployment, as he doesn't know and is not shown what is currently being deployed and which apps would be affected in which way by cancelling the deployment. He would at least need a list of queued deployment steps.
3. _Unwanted states_ is by far not enough information on what this could mean and what side effects there could be. Forcefully stopping a deployment means that all deployments that are conflicting with the one in question will also be stopped. There should be a link in the dialog leading to an explanation of what stopping a deployment effectively means.

As an example for 3.:
- imagine Marathon is deploying a group of three apps (A, B, C), and tasks for A, B, C are waiting to be deployed
- When I restart App A and forcefully _stop the deployment_, the deployment of tasks for apps B and C will also be stopped
- The resulting state will probably be that I have a (probably incorrect number of) tasks of different versions running for B and C

Proposal:
- make the message unambiguous
- add a `more info`, `help`, or a link to an internally served page explaining what forcefully stopping a deployment means.

**Comment by pierlo-upitup** [comment]
cc @leemunroe

**Comment by leemunroe** [comment]
Totally agree. 

@meichstedt You've done a great job here of explaining this. Any suggestions on how to articulate this in a dialog message? We could also ask @sascala nicely to help us out with wording and a docs page.

**Comment by sascala** [comment]
Happy to help. Please create a DOCS Jira and assign it to me.

**Comment by meichstedt** [comment]
@leemunroe Ok, I'll make something up with @sascala's help. Maybe she has an idea of how to wrap my novel into a one-liner :grin:

**Comment by meichstedt** [comment]
Note: This issue has been migrated to https://jira.mesosphere.com/browse/MARATHON-2783. For more information see https://groups.google.com/forum/#!topic/marathon-framework/khtvf-ifnp8.
